python/dgl/nn/pytorch/conv/dotgatconv.py

```diff
@@ -84,8 +95,7 @@ def forward(self, graph, feat):
         graph.edata['sa'] = edge_softmax(graph, graph.edata['a'])
 
         # Step 3. Broadcast softmax value to each edge, and then attention is done
-        graph.apply_edges(lambda edges: {'attn': edges.src['ft'] * \
-                                                 edges.data['sa'].unsqueeze(dim=0).T})
+        graph.apply_edges(fn.u_mul_e('ft', 'sa', 'attn'))
```

Comment:
Please merge this step with the following step by:
```python
graph.update_all(fn.u_mul_e('ft', 'sa', 'attn'), fn.sum('attn', 'agg_u'))
```
to remove the overhead of materializing ft * sa on edges.